src/Nethermind/Nethermind.Network/PeerPool.cs

```diff
@@ -83,6 +87,13 @@ public Peer GetOrAdd(NetworkNode node)
 
         private Peer CreateNew(PublicKeyAsKey key, (Node Node, ConcurrentDictionary<PublicKeyAsKey, Peer> Statics) arg)
         {
+            string enodeString = arg.Node.ToString(Node.Format.ENode);
+
+            if (_trustedNodesManager.IsTrusted(enodeString))
+            {
+                arg.Node.IsTrusted = true;
+            }
+
```

Comment:
I think this logic should be in `PeerManager` or `NodesLoader` rather than in `PeerPool`?

Reply:
Maybe we can avoid it completely as the nodes from source are already marked as trusted?